### PR TITLE
added deletePreparedTransactions method

### PR DESCRIPTION
**Pull request by mrrovot**
solves part of issue #83

### CODE REVIEW

File: lib/WalletBackend.ts [hunk 1]
```diff
@@ -2502,6 +2502,17 @@ export class WalletBackend extends EventEmitter {
         return this.preparedTransactions.delete(transactionHash);
     }
 
+
+    public deletePreparedTransactions(): boolean {
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2506]: Can you add a docstring to this method like the other functions have?

Comment (mrrovot) [reply lib/WalletBackend.ts:2506]: yes, will add

File: lib/WalletBackend.ts [hunk 2]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
         return this.preparedTransactions.delete(transactionHash);
     }
 
+     /**
+     * Deletes all prepared transactions.
+     *
+     * Example:
+     * ```javascript
+     * const deletePreparedTransactions = await wallet.deletePreparedTransactions();
```
Comment (zpalmtree) [inline lib/WalletBackend.ts]:
```suggestion
     * wallet.deletePreparedTransactions();
```

File: lib/WalletBackend.ts [hunk 3]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+     /**
+     * Lists all prepared transactions.
+     *
+     * Example:
+     * ```javascript
+     * const preparedTransactions = await wallet.listPreparedTransactions();
+     * ```
+     *
+     */
+    public listPreparedTransactions(): string[] {
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2533]: Lets rename this `getPreparedTransactions` to match similar functions.

File: lib/WalletBackend.ts [hunk 4]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+     /**
+     * Deletes all prepared transactions.
+     *
+     * Example:
+     * ```javascript
+     * const deletePreparedTransactions = wallet.deletePreparedTransactions();
+     * ```
+     *
+     */
+    public deletePreparedTransactions(): boolean {
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2514]:
```suggestion
    public deletePreparedTransactions(): void {
```

File: lib/WalletBackend.ts [hunk 5]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+        logger.log(
+            'Function deletePreparedTransactions called',
+            LogLevel.DEBUG,
+            LogCategory.GENERAL,
+        );
+
+        return this.preparedTransactions.clear();
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2521]:
```suggestion
        this.preparedTransactions.clear();
```

File: lib/WalletBackend.ts [hunk 6]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+     *
+     */
+    public getPreparedTransactions(): string[] {
+        logger.log(
+            'Function deletePreparedTransactions called',
+            LogLevel.DEBUG,
+            LogCategory.GENERAL,
+        );
+
+        this.preparedTransactions.values();
```
Comment (zpalmtree) [inline lib/WalletBackend.ts]:
```suggestion
        return this.preparedTransactions.values();
```

File: lib/WalletBackend.ts [hunk 7]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+
+        this.preparedTransactions.clear();
+    }
+
+     /**
+     * Gets all prepared transactions.
+     *
+     * Example:
+     * ```javascript
+     * const preparedTransactions = wallet.listPreparedTransactions();
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2529]:
```suggestion
     * const preparedTransactions = wallet.getPreparedTransactions();
```

File: lib/WalletBackend.ts [hunk 8]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+     * ```
+     *
+     */
+    public getPreparedTransactions(): string[] {
+        logger.log(
+            'Function deletePreparedTransactions called',
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2535]:
```suggestion
            'Function getPreparedTransactions called',
```

File: lib/WalletBackend.ts [hunk 9]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
         return this.preparedTransactions.delete(transactionHash);
     }
 
+     /**
+     * Deletes all prepared transactions.
+     *
+     * Example:
+     * ```javascript
+     * const deletePreparedTransactions = wallet.deletePreparedTransactions();
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2510]:
```suggestion
     * wallet.deletePreparedTransactions();
```

File: lib/WalletBackend.ts [hunk 10]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+     *
+     */
+    public getPreparedTransactions(): string[] {
+        logger.log(
+            'Function getPreparedTransactions called',
+            LogLevel.DEBUG,
+            LogCategory.GENERAL,
+        );
+
+        return this.preparedTransactions.values();
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2540]:
```suggestion
        return Array.from(this.preparedTransactions.values());
```

`.values()` returns an iterator, which we need to convert into an array.

File: lib/WalletBackend.ts [hunk 11]
```diff
@@ -2502,6 +2502,44 @@ export class WalletBackend extends EventEmitter {
...
+            'Function deletePreparedTransactions called',
+            LogLevel.DEBUG,
+            LogCategory.GENERAL,
+        );
+
+        this.preparedTransactions.clear();
+    }
+
+     /**
+     * Gets all prepared transactions.
```
Comment (zpalmtree) [inline lib/WalletBackend.ts:2525]:
```suggestion
     * Gets all prepared transaction hashes.
```

**Comment by zpalmtree** [comment]
Merged in 2dbc60bd7f139c8343191969595b7150df8efd05

**Pull request closed without merging**